src/mavsdk/core/mavlink_parameters.cpp

```diff
@@ -1,58 +1,86 @@
 #include "mavlink_parameters.h"
+#include "mavlink_message_handler.h"
+#include "timeout_handler.h"
 #include "system_impl.h"
 #include <algorithm>
 #include <cstring>
 #include <future>
 
 namespace mavsdk {
 
-MAVLinkParameters::MAVLinkParameters(SystemImpl& parent) : _parent(parent)
-{
-    _parent.register_mavlink_message_handler(
-        MAVLINK_MSG_ID_PARAM_VALUE,
-        [this](const mavlink_message_t& message) { process_param_value(message); },
-        this);
+MAVLinkParameters::MAVLinkParameters(
+    Sender& sender,
+    MavlinkMessageHandler& message_handler,
+    TimeoutHandler& timeout_handler,
+    TimeoutSCallback timeout_s_callback,
+    bool is_server) :
+    _sender(sender),
+    _message_handler(message_handler),
+    _timeout_handler(timeout_handler),
+    _timeout_s_callback(timeout_s_callback),
+    _is_server(is_server)
+{
+    if (const char* env_p = std::getenv("MAVSDK_PARAMETER_DEBUGGING")) {
+        if (std::string(env_p) == "1") {
+            LogDebug() << "Parameter debugging is on.";
+            _parameter_debugging = true;
+        }
+    }
 
-    _parent.register_mavlink_message_handler(
-        MAVLINK_MSG_ID_PARAM_SET,
-        [this](const mavlink_message_t& message) { process_param_set(message); },
-        this);
+    if (!_is_server) {
+        _message_handler.register_one(
+            MAVLINK_MSG_ID_PARAM_VALUE,
+            [this](const mavlink_message_t& message) { process_param_value(message); },
+            this);
 
-    _parent.register_mavlink_message_handler(
-        MAVLINK_MSG_ID_PARAM_EXT_VALUE,
-        [this](const mavlink_message_t& message) { process_param_ext_value(message); },
-        this);
+        //_message_handler.register_one(
+        //    MAVLINK_MSG_ID_PARAM_SET,
+        //    [this](const mavlink_message_t& message) { process_param_set(message); },
+        //    this);
```

Comment:
Should that be removed or uncommented?

Reply:
Thanks, I believe this is only used for when it's a server. I'm removing it.